tianshou/highlevel/experiment.py

```diff
@@ -491,14 +491,26 @@ def _get_optim_factory(self) -> OptimizerFactory:
         else:
             return self._optim_factory
 
-    def build(self, add_seeding_info_to_name: bool = False) -> Experiment:
-        """Creates the experiment based on the options specified via this builder.
+    def build(
+        self,
+        num_experiments: int = 1,
+        add_seeding_info_to_name: bool = False,
+    ) -> list[Experiment]:
```

Comment:
The change in the return is breaking. We could make it backwards compatible by returning `Experiment | list[Experiment]`. While it's somehow less elegant, at the same time `num_experiments=1` is an important special case, and it might be a bit annoying to always have to use `build()[0]` for it.

Alternatively, or even additionally, we can adjust the launcher interface to just call `.run()` in the main process in case a list with a single entry is passed. @opcode81 wdyt?

Reply:
I think making any launcher just run in the main process if a single experiment is passed is a good approach. It would be easy to implement: make a refactoring renaming `launch` to `_launch` in the base class, and then add an actual implementation of the public `launch` in the base class that either calls the abstract `_launch` or calls `experiments[0].run()`

Reply:
Doesn't fully resolve the breaking situation (let's wait for @opcode81's view on this), but at least improves a bit. Then one can just always use the launcher